### PR TITLE
SDK-303 [Python SDK] Allow oAuth2 token Storage initialization

### PR DESCRIPTION
**Pull request by artemgolozubov**
Pull Request Checklist
======================

Please read this for our PR Culture:
------------------------------------
https://incountry.atlassian.net/wiki/spaces/ED/pages/46858869/Pull+requests+culture+Draft

Work relates to:
----------------
Link(s) to Jira Task/User Story/Bug/Epic:
- [SDK-303](https://incountry.atlassian.net/browse/SDK-303)

Code Quality (Before submitting PR)
------------
- [x] Ensure code compiles and passes Sonar Lint.
- [x] Do your best to ensure integration tests pass. If they don't then...
- [x] Remove, ignore, or mark pending any brittle tests.
- [x] Remove commented out code.

Target Release
--------------
- [x] Please put the target release vehicle this needs to be bundled with.
- [x] Did you increment the version? What is the new version?

### CODE REVIEW

File: incountry/models/storage_with_env.py [hunk 1]
```diff
@@ -80,22 +81,34 @@ def validate_secret_key_accessor(cls, value, values):
...
+                f"Please choose either API key authorization or oAuth (client_id + client_secret/oauth_token) "
+                f"authorization, not both"
+            )
 
-        if values["api_key"] is not None and (values["client_id"] is not None or values["client_secret"] is not None):
+        if oauth_token is not None and has_oauth_creds:
             raise ValueError(
-                f"Please choose either API key authorization or oAuth (client_id + client_secret) authorization, "
-                f"not both"
+                f"Please choose either authorization with previously acquired oAuth token or oAuth authorization by "
```
Comment (RommeroInCountry) [inline incountry/models/storage_with_env.py:98]: authorization using oAuth token or authorization using oAuth credentials (client_id + client_secret), not both

**Review by RommeroInCountry — CHANGES_REQUESTED** [review]
let's move this PR further and include API Key removal as well. 
We can skip 3.5.0 release in favour of 4.0.0

**Review by RommeroInCountry — CHANGES_REQUESTED** [review]

File: incountry/storage.py [hunk 2]
```diff
@@ -57,15 +57,15 @@ def __init__(
...
-                Your API key. Defaults to None.
-                Can also be set via INC_API_KEY environment variable.
             client_id:
                 Client Id used for oAuth authorization. Defaults to None.
                 Can also be set via INC_CLIENT_ID environment variable.
             client_secret:
                 Client Secret used for oAuth authorization. Defaults to None.
                 Can also be set via INC_CLIENT_SECRET environment variable.
+            oauth_token:
+                OAuth token which is already acquired prior to Storage initialization.
```
[the remark this answers is not in the document]
Comment (artemgolozubov) [reply incountry/storage.py:67]: @RommeroInCountry It's just a copy-paste from Node SDK Readme file. See https://github.com/incountry/sdk-nodejs/commit/c7ae1867dab570fc34846703fe1481605c3e290e#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R50
Are you sure we need to change this text here?

[the remark this answers is not in the document]
Comment (RommeroInCountry) [reply incountry/storage.py:67]: Then we'd better update text in Node SDK as well :D it's just to excessive right now

**Review by RommeroInCountry — DISMISSED** [review]

File: incountry/models/storage_with_env.py [hunk 3]
```diff
@@ -80,22 +80,25 @@ def validate_secret_key_accessor(cls, value, values):
...
+        client_secret = values.get("client_secret", None)
+        oauth_token = values.get("oauth_token", None)
+        has_oauth_creds = client_id is not None or client_secret is not None
 
-        if values["api_key"] is not None and (values["client_id"] is not None or values["client_secret"] is not None):
+        if oauth_token is not None and has_oauth_creds:
             raise ValueError(
-                f"Please choose either API key authorization or oAuth (client_id + client_secret) authorization, "
-                f"not both"
+                f"Please choose either authorization using oAuth token or authorization using oAuth "
```
Comment (igor-alekseev-dev) [inline incountry/models/storage_with_env.py:90]: I think this needs to be rephrased

**Review by igor-alekseev-dev — APPROVED** [review]

**Review by RommeroInCountry — APPROVED** [review]

**Pull request merged**